src/main/scala/WeaviateDataWriter.scala

```diff
@@ -34,8 +35,19 @@ case class WeaviateDataWriter(weaviateOptions: WeaviateOptions, schema: StructTy
         writeBatch(retries - 1)
       }
     }
-    logInfo(s"Writing batch successful. IDs of inserted objects: ${IDs}")
-    batch.clear()
+
+    val (objectsSuccess, objectsWithError) = results.getResult.partition(_.getResult.getErrors == null)
```

Comment:
Is this meant to be `objectsWithSuccess`?

Reply:
that does seem to sound better and consistent although I did purposely call it objectsSuccess. Let me change it to `objectsWithSuccess`